cmd/notify-webhook.go

```diff
@@ -48,33 +50,119 @@ type httpConn struct {
 	Endpoint string
 }
 
-// Lookup endpoint address by successfully dialing.
-func lookupEndpoint(u *url.URL) error {
-	dialer := &net.Dialer{
-		Timeout:   300 * time.Millisecond,
-		KeepAlive: 300 * time.Millisecond,
+// List of success status.
+var successStatus = []int{
+	http.StatusOK,
+	http.StatusAccepted,
+	http.StatusContinue,
+	http.StatusNoContent,
+	http.StatusPartialContent,
+}
+
+// isNetErrorRetryable - is network error retryable.
+func isNetErrorRetryable(err error) bool {
+	if err == nil {
+		return false
+	}
+	if strings.Contains(err.Error(), "Client.Timeout exceeded while awaiting headers") {
+		return true
 	}
-	nconn, err := dialer.Dial("tcp", canonicalAddr(u))
+	switch err.(type) {
+	case net.Error:
+		switch err.(type) {
+		case *net.DNSError, *net.OpError, net.UnknownNetworkError:
+			return true
+		case *url.Error:
+			// For a URL error, where it replies back "connection closed"
+			// retry again.
+			if strings.Contains(err.Error(), "Connection closed by foreign host") {
+				return true
+			}
+		default:
+			if strings.Contains(err.Error(), "net/http: TLS handshake timeout") {
+				// If error is - tlsHandshakeTimeoutError, retry.
+				return true
+			} else if strings.Contains(err.Error(), "i/o timeout") {
+				// If error is - tcp timeoutError, retry.
+				return true
+			} else if strings.Contains(err.Error(), "connection timed out") {
+				// If err is a net.Dial timeout, retry.
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Lookup endpoint address by successfully dialing.
+func lookupEndpoint(urlStr string) error {
+	minioRelease := bytes.NewReader([]byte(fmt.Sprintf("{\"release\": %s", ReleaseTag)))
+	req, err := http.NewRequest("POST", urlStr, minioRelease)
 	if err != nil {
 		return err
 	}
-	return nconn.Close()
+
+	client := &http.Client{
+		Timeout: 1 * time.Second,
+		Transport: &http.Transport{
+			// need to close connection after usage.
+			DisableKeepAlives: true,
+		},
+	}
+
+	// Set content-type.
+	req.Header.Set("Content-Type", "application/json")
+
+	// Set proper server user-agent.
+	req.Header.Set("User-Agent", globalServerUserAgent)
+
+	for {
```

Comment:
Looks like 'for' here is only needed to follow redirect urls, can you also add a comment here ?